subset/pentests/brute_client.py

```diff
@@ -0,0 +1,79 @@
+import socket, sys, json
+
+user_data=open('user_list.json')
+user_list = json.load(user_data)
+user_data.close()
+
+#username=['root','admin','manager']
+username= user_list['username'].split(",")
+
+#password=['root','pass','friend']
+password= user_list['password'].split(",")
+
+arguments = sys.argv
+
+print 'brute_client test begin'
+print 'address:' + arguments[1]
+print 'port:' + arguments[2]
+print 'reportFile:' + arguments[3]
+
+#address = '127.0.0.1'
+address = str(arguments[1])
+
+#port = 10000
+port = int(arguments[2])
+
+report_file = str(arguments[3])
+data_length =64
+
+expected=['login:','Password:','Last login:','Login incorrect','Connection closed by foreign host.']
+expected1=['username:','password:','success','login failed','Connection closed by foreign host.']
+
+sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+server_address = (address,port)
+
+sock.connect(server_address)
+
+login_report='\n'
+u=0
+p=0
+
+def _write_report():
+    file = open(report_file, "w")
+    file.write(login_report)
+
+try:
+    while True:
+        data = sock.recv(data_length)
+        if data:
+            print data
+            if data == expected[2]:
+                #success statement
+                login_report = login_report + 'Username:' + username[u-1] + '\n'
+                login_report = login_report + 'Password:' + password[p-1] + '\n'
+                _write_report()
```

Comment:
I don't understand why you have to call _write_report() from 4 places, rather than just once after the loop exits. Also, why is this the only case that calls sock.close()?

Reply:
There are two cases where I don't want the report written to because the test is still in progress.
sock.close() is only called when finished.
Finished is either a connection drop from the server or a success from the server

Reply:
Bump -- still contend you only need to call _write_report() one time (at the same time you close the socket)

Reply:
The code doesnt break if it still trying username and passwords.

Agreed for the _write_report() sorry code blind and couldn't see it until you pointed the obvious